tests/unit/TextUI/SourceFilterTest.php

```diff
@@ -9,256 +9,422 @@
+                        ],
+                    ),
+                ),
+            ],
+            'globstar includes all files' => [
+                [
+                    self::fixturePath('a/c/d/PrefixSuffix.php') => true,
+                    self::fixturePath('a/c/PrefixSuffix.php')   => true,
+                    self::fixturePath('a/PrefixSuffix.php')     => true,
+                ],
+                self::createSource(
+                    includeDirectories: FilterDirectoryCollection::fromArray(
                         [
                             new FilterDirectory(
-                                $fixtureDirectory,
+                                self::fixturePath('**'),
                                 '',
                                 '.php',
                             ),
                         ],
                     ),
-                    FileCollection::fromArray([]),
-                    FilterDirectoryCollection::fromArray([]),
-                    FileCollection::fromArray([]),
-                    false,
-                    false,
-                    false,
-                    false,
-                    false,
-                    false,
-                    false,
-                    false,
-                    false,
-                    [
-                        'functions' => [],
-                        'methods'   => [],
-                    ],
-                    false,
-                    false,
-                    false,
                 ),
             ],
-            'file included using directory, but excluded using file' => [
-                false,
-                $fixtureDirectory . DIRECTORY_SEPARATOR . 'a' . DIRECTORY_SEPARATOR . 'PrefixSuffix.php',
-                new Source(
-                    null,
-                    false,
-                    FilterDirectoryCollection::fromArray(
+            'globstar with any single char prefix includes sibling files' => [
+                [
+                    self::fixturePath('a/PrefixSuffix.php')     => false,
+                    self::fixturePath('a/c/PrefixSuffix.php')   => true,
+                    self::fixturePath('a/c/d/PrefixSuffix.php') => true,
+                ],
+                self::createSource(
+                    includeDirectories: FilterDirectoryCollection::fromArray(
                         [
                             new FilterDirectory(
-                                $fixtureDirectory,
+                                self::fixturePath('a/c/Z**'),
                                 '',
                                 '.php',
                             ),
                         ],
                     ),
-                    FileCollection::fromArray([]),
-                    FilterDirectoryCollection::fromArray([]),
-                    FileCollection::fromArray(
+                ),
+            ],
+            'globstar with any more than a single char prefix does not include sibling files' => [
```

Comment:
this behavior is very strange. `/<any char>**` will be equivilent to `/*` but `<more than 1 char>**` will not